### PR TITLE
mostly moving files around + fix the sporadic job cancelling

### PR DESCRIPTION
**Pull request by mordamax**
- Split huge file by methods into its own files so it's easier later to simplify even more + cover these methods with tests,
- Fix https://github.com/paritytech/command-bot/issues/146 - removed pre-commit
- Fix https://github.com/paritytech/command-bot/issues/103 sporadic cancelling of the jobs - https://github.com/paritytech/command-bot/pull/147/files#diff-1aaca4f4a94cd93a077fa48866e4d2df649fb0d090d2a4bd19df798b785cbae3R26 
the reason was: when you run few commands with one comment (https://github.com/paritytech-stg/polkadot/pull/2#issuecomment-1372914387), it was too quick force pushing the new gitlab-ci.yml config to the same branch, so the gitlab pipeline didn't have a time to read&start a pipeline, while it was already overridden by next command
Now jobs are set in queue as they should having the task ID as variation additionally to the PR number
image

### CODE REVIEW

File: src/bot/parsePullRequestBotCommandLine.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import assert from "assert"
```
Comment (mordamax) [inline src/bot/parsePullRequestBotCommandLine.ts:1]: moved from bot.ts

File: src/bot/setupEvent.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { Probot } from "probot"
```
Comment (mordamax) [inline src/bot/setupEvent.ts:1]: moved from bot.ts

File: src/bot/types.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { EmitterWebhookEventName } from "@octokit/webhooks/dist-types/types"
```
Comment (mordamax) [inline src/bot/types.ts:1]: moved from bot.ts

File: src/gitlab.ts [hunk 4]
```diff
@@ -7,20 +7,31 @@ import path from "path"
...
 // Integration tests don't like waiting for 16 seconds
 const pipelineUpdateInterval = process.env.GITLAB_PIPELINE_UPDATE_INTERVAL
   ? envNumberVar("GITLAB_PIPELINE_UPDATE_INTERVAL")
   : 16384
 
-const getPipelineScriptsCloneCommand = ({ withRef }: { withRef: boolean }) =>
-  `git clone --progress --verbose --depth 1 ${
-    withRef ? `--branch "$PIPELINE_SCRIPTS_REF"` : ""
-  } "$PIPELINE_SCRIPTS_REPOSITORY" "$PIPELINE_SCRIPTS_DIR"`
+function getCiBranchName(task: Task): string {
```
Comment (mordamax) [inline src/gitlab.ts:26]: this actually fixes the sporadic cancelling

File: src/gitlab.ts [hunk 5]
```diff
@@ -172,68 +180,6 @@ export const runCommandInGitlabPipeline = async (ctx: Context, task: Task): Prom
   return getAliveTaskGitlabContext(ctx, { id: pipeline.id, projectId: pipeline.project_id, jobWebUrl: job.web_url })
 }
 
-export function getGitlabCiYmlConfig(
```
Comment (mordamax) [inline src/gitlab.ts:175]: moved to its own file

File: src/bot/events/onIssueCommentCreated.ts [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+import { parsePullRequestBotCommandLine } from "src/bot/parsePullRequestBotCommandLine"
+import { ParsedBotCommand, WebhookHandler } from "src/bot/types"
+import { isRequesterAllowed } from "src/core"
+import { getSortedTasks } from "src/db"
+import { createComment, getPostPullRequestResult, updateComment } from "src/github"
+import { cancelTask, getNextTaskId, PullRequestTask, queueTask, serializeTaskQueuedDate } from "src/task"
+import { PullRequestError } from "src/types"
+import { getLines } from "src/utils"
+
+export const onIssueCommentCreated: WebhookHandler<"issue_comment.created"> = async (ctx, octokit, payload) => {
```
Comment (mordamax) [inline src/bot/events/onIssueCommentCreated.ts:13]: moved from bot.ts

File: Dockerfile [hunk 7]
```diff
@@ -46,15 +46,10 @@ CMD yarn start
...
-    python3 \
     make \
     bash \
     sed \
-    python3-distutils \
-    python3-pkg-resources \
     gcc \
-    python3-dev \
     libc-dev && \
     apt-get autoremove -y
```
Comment (mutantcornholio) [inline Dockerfile:55]: This all can fit nicely into the base image, having a separate CI image doesn't make much sense anymore

File: src/commands.ts [hunk 8]
```diff
@@ -10,6 +10,7 @@ import { CmdJson } from "src/schema/schema.cmd"
 import { CommandRunner } from "src/shell"
 import { CommandConfigs } from "src/types"
 
+export const PIPELINE_SCRIPTS_REF = "PIPELINE_SCRIPTS_REF"
```
Comment (mutantcornholio) [inline src/commands.ts:13]: What does this do?

File: src/gitlab/createCiConfig.ts [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        "fi",
+        `export ARTIFACTS_DIR="$PWD/${artifactsFolderPath}"`,
+        /*
+          The artifacts directory might be left over from a previous run in
+          the same Gitlab shell executor
+        */
+        'rm -rf "$ARTIFACTS_DIR"',
+        'mkdir -p "$ARTIFACTS_DIR"',
+        task.command,
+      ],
```
Comment (mutantcornholio) [inline src/gitlab/createCiConfig.ts:44]:
```suggestion
export function createCiConfig(
  headSha: string,
  task: Task,
  pipelineScripts: PipelineScripts,
  jobTaskInfoMessage: string,
): object {
  const artifactsFolderPath = ".git/.artifacts"
  return {
    workflow: { rules: [{ if: `$CI_PIPELINE_SOURCE == "api"` }, { if: `$CI_PIPELINE_SOURCE == "web"` }] },
    command: {
      timeout: "24 hours",
      ...task.gitlab.job,
      script: [
        `
        echo "This job is related to task ${task.id}. ${jobTaskInfoMessage}."

        # The scripts repository might be left over from a previous run in the
        # same Gitlab shell executor

        rm -rf "$PIPELINE_SCRIPTS_DIR"
        if [ "\${PIPELINE_SCRIPTS_REPOSITORY:-}" ]; then
          if [ "\${PIPELINE_SCRIPTS_REF:-}" ]; then
            git clone --progress --verbose --depth 1 --branch "$PIPELINE_SCRIPTS_REF" "$PIPELINE_SCRIPTS_REPOSITORY" "$PIPELINE_SCRIPTS_DIR"
          else
            git clone --progress --verbose --depth 1 "$PIPELINE_SCRIPTS_DIR"
          fi
        fi
        export ARTIFACTS_DIR="$PWD/${artifactsFolderPath}"
        # The scripts repository might be left over from a previous run in the
        # same Gitlab shell executor

        rm -rf "$PIPELINE_SCRIPTS_DIR"
        mkdir -p "$ARTIFACTS_DIR"
        `,
        task.command,
      ],
```

File: src/commands.ts (hunk 8 above)
Comment (mordamax) [reply src/commands.ts:13]: just having a constant (for autocomplete) vs hardcoding as string

File: Dockerfile (hunk 7 above)
Comment (mordamax) [reply Dockerfile:55]: @alvicsam recommended to leave as is, he will clean this up in https://github.com/paritytech/ci_cd/issues/690 💪

**Review by mutantcornholio — APPROVED** [review]

**Pull request merged**